### DISCUSSION TITLE
Remembered window size should always be smaller than maximized size

### DISCUSSION THREAD

**Discussion by qwertychouskie** [rxhanson/Rectangle#1371] [General]
Sometimes it happens that the maximized window size becomes the remembered window size, so when un-maximizing it, it stays the same size.  My suggestion would be to cap the remembered un-maximized size to e.g. 80% of the maximized size, it would make things feel less janky.

**Comment by rxhanson** [comment]
Thanks for the feedback. I don't think I'll personally add this in, but would be fine merging a good pull request that adds this in as a Terminal command configuration.

**Reply by qwertychouskie** [reply]
For what it's worth, this is how Gnome handles it out of the box:

https://github.com/rxhanson/Rectangle/assets/13989090/f6202575-5e2b-4715-8bc8-a52acf60ba55

I would argue Gnome is a pretty good reference point for a macOS-like interface but with window management handled properly, so doing the same thing with Rectangle by default is probably not a terrible idea.  It's kinda hard to describe why it matters so much, but in my experience it goes a *long* way towards making things feel sane and polished.

I swear the moment the EU forces Apple to make iOS app compiling/testing on other OSes possible, this laptop is getting Asahi, but in the mean time, I sadly have to make do with "Gnome but if it had a bunch of legacy design decisions from 15-30 years ago still sticking around and making things miserable".  In the mean time, I'm just stuck trying to patch in sane behavior via oodles of 3rd-party programs.  

Thanks for making one of those 3rd-party programs that helps make things bearable.

p.s. BTW, if you have any idea on how to make the Enter key in Finder work as one would expect (to ENTER a file/folder, NOT TO RENAME, APPLE), that doesn't require disabling SIP like XtraFinder does (I tried disabling SIP and it just gave a cryptic error that no one online seems to know how to fix), please let me know ASAP.  In the mean time, maybe I'll keep trying to find a way to make a modern version of Nautilus run (MacPorts' version is like 5 years old).
